cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/common/DBRecord.java

```diff
@@ -0,0 +1,146 @@
+
+  /**
+   * @return the {@link StructuredRecord} contained in this object
+   */
+  public StructuredRecord getRecord() {
+    return record;
+  }
+
+  /**
+   * Builds the {@link #record} using the specified {@link ResultSet}
+   *
+   * @param resultSet the {@link ResultSet} to build the {@link StructuredRecord} from
+   */
+  public void readFields(ResultSet resultSet) throws SQLException {
+    List<Schema.Field> fields = Lists.newArrayList();
+    ResultSetMetaData metadata = resultSet.getMetaData();
+    // ResultSetMetadata columns are numbered starting with 1
+    for (int i = 1; i <= metadata.getColumnCount(); i++) {
+      String columnName = metadata.getColumnName(i);
+      String columnType = metadata.getColumnClassName(i);
+      Schema.Field field = Schema.Field.of(columnName, Schema.of(getType(columnType)));
+      fields.add(field);
+    }
+    Schema schema = Schema.recordOf("dbRecord", fields);
+    StructuredRecord.Builder recordBuilder = StructuredRecord.builder(schema);
+    for (Schema.Field field : fields) {
+      recordBuilder.set(field.getName(), resultSet.getObject(field.getName()));
+    }
+    record = recordBuilder.build();
+  }
+
+  public void write(DataOutput out) throws IOException {
+  }
+
+  /**
+   * Writes the {@link #record} to the specified {@link PreparedStatement}
+   *
+   * @param stmt the {@link PreparedStatement} to write the {@link StructuredRecord} to
+   */
+  public void write(PreparedStatement stmt) throws SQLException {
+    Schema recordSchema = record.getSchema();
+    List<Schema.Field> schemaFields = recordSchema.getFields();
+    for (int i = 0; i < schemaFields.size(); i++) {
+      // In JDBC, field indices start with 1
+      int jdbcFieldIndex = i + 1;
+      Schema.Field field = schemaFields.get(i);
+      String fieldName = field.getName();
+      Schema.Type fieldType = field.getSchema().getType();
+      Object fieldValue = record.get(fieldName);
+      writeValue(stmt, fieldType, fieldValue, jdbcFieldIndex);
+    }
+  }
+
+  private Schema.Type getType(String columnType) {
+    if (columnType == null) {
+      return Schema.Type.NULL;
+    } else if (String.class.getName().equals(columnType)) {
+      return Schema.Type.STRING;
+    } else if (Boolean.class.getName().equals(columnType)) {
+      return Schema.Type.BOOLEAN;
+    } else if (Integer.class.getName().equals(columnType)) {
+      return Schema.Type.INT;
+    } else if (Long.class.getName().equals(columnType)) {
+      return Schema.Type.LONG;
+    } else if (Float.class.getName().equals(columnType)) {
+      return Schema.Type.FLOAT;
+    } else if (Double.class.getName().equals(columnType)) {
+      return Schema.Type.DOUBLE;
+    } else {
+      throw new IllegalArgumentException("Unsupported datatype: " + columnType);
+    }
+  }
+
+  private void writeValue(PreparedStatement stmt, Schema.Type fieldType, Object fieldValue,
+                          int fieldIndex) throws SQLException {
+    switch (fieldType) {
```

Comment:
Only reason was that we use enums here and strings in the other method. maybe I can use ``if`` in both places.